### PR TITLE
fix: set kafka_skip_broken_messages on dead letter queue table

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Problem

#9752

## Changes

Set `kafka_skip_broken_messages` on the DLQ table.

This should have went in a long time ago via a PR I opened a long time ago.

## How did you test this code?

Ran the migration locally, checked the schema in ClickHouse

### CODE REVIEW

File: ee/clickhouse/sql/dead_letter_queue.py [hunk 1]
```diff
@@ -46,7 +46,12 @@
     ttl_period=ttl_period("_timestamp", 4),  # 4 weeks
 )
 
-KAFKA_DEAD_LETTER_QUEUE_TABLE_SQL = lambda: DEAD_LETTER_QUEUE_TABLE_BASE_SQL.format(
+# skip up to 1000 messages per block. blocks can be as large as 65505
+# if a block has >1000 broken messages it probably means we're doing something wrong
```
Comment (tiina303) [inline ee/clickhouse/sql/dead_letter_queue.py:50]: or someone could be DDOS'ing us? Is there a graph that will show broken messages if we skip them?

**Review by tiina303 — APPROVED** [review]
Thanks 🙏

File: ee/clickhouse/sql/dead_letter_queue.py (hunk 1 above)
Comment (yakkomajuri) [reply ee/clickhouse/sql/dead_letter_queue.py:50]: well in that case we should be made aware too

**Comment by fuziontech** [comment]
Can we make the DLQ table _not_ `JSONEachRow`? This would make things a lot easier and more robust. Basically just capture the string or bytes as a single column. We really shouldn't be trying to deserialize something that is in the DLQ because 9/10 times the reason why it will be in the DLQ will be a serialization issue.

This would be better than dropping the payload on the floor since one of the reasons to have a DLQ is for investigative purposes. If we just skip the message we lose any intel we can use to fix the problem in the future.

**Comment by yakkomajuri** [comment]
@fuziontech fair point. what I do like about the current structure is that it makes it easy to re-ingest events if it's "our fault" (plugin server issue, Postgres down, etc). But yeah I see how it can be problematic. 

I think we can keep columns we have some control over and can do basic validation on (e.g. team_id, uuid), and then keep the rest as an arbitrary payload. Completely getting rid of JSONEachRow would mean not having things like error message, location, and timestamp, which we have full control over and are valuable

**Comment by yakkomajuri** [comment]
But yeah worth doing on a follow up

**Pull request merged**